test/algo/compression/v2/test_auto_conv.py

```diff
@@ -0,0 +1,154 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import unittest
+
+import torch
+import types
+import torch.nn.functional as F
+
+from nni.compression.pytorch.pruning import LevelPruner
+from nni.algorithms.compression.v2.pytorch.utils import compute_sparsity_mask2compact
+
+class CondModel(torch.nn.Module):
+    """
+    test for:
+        prim::If
+    """
+    the_cond: bool
+    def __init__(self):
+        super().__init__()
+        self.the_cond = True
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        if self.the_cond:
+            x = x + 0.00001
+        else:
+            x = x - 0.00001
+        self.the_cond = not self.the_cond
+        return x
+
+class ASubModel(torch.nn.Module):
+    """
+    test for:
+        sub model
+    """
+    def __init__(self):
+        super().__init__()
+
+    def forward(self, x: torch.Tensor) -> torch.Tensor:
+        x = x + 0.00001
+        return x
+
+class TorchModel1(torch.nn.Module):
+    """
+    test for:
+        add, sub, mul, div, exp, matmul,
+        relu, gelu, tanh, silu, sigmod, softmax,
+        size, unsqueeze, flatten, cat, slice, reshape, transpose, t, select, permute, constant_pad_nd,
+        mean, avg_pool2d, max_pool2d, sum, adaptive_avg_pool2d,
+        to, Int, view,
+        type_as, expand_as, contiguous,
+
+    notes:
+        'floor_divide' have no backward, then not be tested
+    """
+    def __init__(self):
+        super().__init__()
+        self.conv1 = torch.nn.Conv2d(1, 6, 5, 1)
+        self.conv2 = torch.nn.Conv2d(6, 16, 5, 1)
+        self.fccond = torch.nn.Linear(16 * 4 * 4, 16 * 4 * 4)
+        self.fc1 = torch.nn.Linear(16 * 4 * 4, 120)
+        self.fc2 = torch.nn.Linear(120, 84)
+        self.fc3 = torch.nn.Linear(84, 10)
+        self.pool1 = torch.nn.MaxPool2d((2, 2))
+        self.pool2 = torch.nn.MaxPool2d((2, 2))
+        self.cond = torch.jit.script(CondModel())
```

Comment:
we can handle `if` by this way?

Reply:
`torch.jit.script` build the graph from the ast, so it'll catch and keep a `prim::If` op to the TorchScript. but the graph_utils.py cannot handle the `prim::If` op now.

Reply:
i think it's a better way to pack all the branch or loop in a bounded leaf module.

Reply:
> i think it's a better way to pack all the branch or loop in a bounded leaf module.

we can discuss how to support `if & loop` in next speedup meeting.